js/react_native/android/srcExt/main/java/ai/onnxruntime/reactnative/OnnxruntimeModule.java

```diff
@@ -0,0 +1,321 @@
+    return "Onnxruntime";
+  }
+
+  /**
+   * React native binding API to load a model using given uri.
+   *
+   * @param uri a model file location. it's used as a key when multiple sessions are created, i.e. multiple models are
+   *     loaded.
+   * @param options onnxruntime session options
+   * @param promise output returning back to react native js
+   * @note when run() is called, the same uri must be passed into the first parameter.
+   */
+  @ReactMethod
+  public void loadModel(String uri, ReadableMap options, Promise promise) {
+    try {
+      WritableMap resultMap = loadModel(uri, options);
+      promise.resolve(resultMap);
+    } catch (Exception e) {
+      promise.reject("Can't load model \"" + uri + "\": " + e.getMessage(), e);
+    }
+  }
+
+  /**
+   * React native binding API to run a model using given uri.
+   *
+   * @param key a model file location given at loadModel()
+   * @param input an input tensor
+   * @param output an output names to be returned
+   * @param options onnxruntime run options
+   * @param promise output returning back to react native js
+   */
+  @ReactMethod
+  public void run(String key, ReadableMap input, ReadableArray output, ReadableMap options, Promise promise) {
+    try {
+      WritableMap resultMap = run(key, input, output, options);
+      promise.resolve(resultMap);
+    } catch (Exception e) {
+      promise.reject("Fail to inference: " + e.getMessage(), e);
+    }
+  }
+
+  /**
+   * Load a model from raw resource directory.
+   *
+   * @param uri uri parameter from react native loadModel()
+   * @param options onnxruntime session options
+   * @return model loading information, such as key, input names, and output names
+   */
+  public WritableMap loadModel(String uri, ReadableMap options) throws Exception {
+    InputStream modelStream = reactContext.getApplicationContext().getContentResolver().openInputStream(Uri.parse(uri));
+    WritableMap resultMap = loadModel(uri, modelStream, options);
+    modelStream.close();
+    return resultMap;
+  }
+
+  /**
+   * Load a model from raw resource directory.
+   *
+   * @param uri uri parameter from react native loadModel() or dummy string for unit testing purpose
+   * @param modelStream a input stream to read a model
+   * @param options onnxruntime session options
+   * @return model loading information, such as key, input names, and output names
+   */
+  public WritableMap loadModel(String uri, InputStream modelStream, ReadableMap options) throws Exception {
+    OrtSession ortSession = null;
+
+    if (!sessionMap.containsKey(uri)) {
+      byte[] modelArray = null;
+
+      Reader reader = new BufferedReader(new InputStreamReader(modelStream));
+      modelArray = new byte[modelStream.available()];
+      modelStream.read(modelArray);
+
+      SessionOptions sessionOptions = parseSessionOptions(options);
+
+      // Optional call of sessionOptions java api to enable usage of ort extensions custom ops when user
```

Comment:
same source files except enabling the registerCustomOps call here. 

but gradle only support selectively including sourceDir. (no sourceFiles support.) so included another set - `srcExt/main` here when user specifying `ortPackageName` field as `onnxruntime-ext`.

Reply:
could we reduce the amount of duplicated code?

e.g., just have two implementations of a method `RegisterOrtExtensionsCustomOpsIfEnabled()` (perhaps there's a better name) and put those in separate files. then we can just call that here and keep a single copy of all the other code.

Reply:
Yes. that's the first approach I tried. but I have to put the two impls under two classes (with different name) in the separate files. and then that lead to two different style calls in OnnxruntimeModule.java i.e.

`OnnxruntimeExtensions.RegisterOrtExtensionsCustomOpsLibrary()`
`OnnxruntimeExtensionsDummy.RegisterOrtExtensionsCustomOpsLibrary()`

Reply:
hm, can we have two different files that contain a class with the same name? e.g., srcExtensionsEnabled/.../className.java and srcExtensionsDisabled/.../className.java.

then we only include one of them in the project

Reply:
ok thanks for the suggestion.

found out a similar way to do this that can work. will push the change.